test/Microsoft.ML.Tests/Transformers/CustomTransformerTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Runtime.Tools;
+using System.ComponentModel.Composition;
+using System.ComponentModel.Composition.Hosting;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests.Transformers
+{
+    public sealed class CustomTransformerTests : TestDataPipeBase
+    {
+        public CustomTransformerTests(ITestOutputHelper helper) : base(helper)
+        {
+        }
+
+        public class MyInput
+        {
+            public float Float1 { get; set; }
+            public float[] Float4 { get; set; }
+        }
+
+        public class MyOutput
+        {
+            public string Together { get; set; }
+        }
+
+        public class MyLambda
+        {
+            [Export("MyLambda")]
+            public ITransformer MyTransformer => new CustomMappingTransformer<MyInput, MyOutput>(ML, MyAction, "MyLambda");
+
+            [Import]
+            public MLContext ML { get; set; }
+
+            public static void MyAction(MyInput input, MyOutput output)
+            {
+                output.Together = $"{input.Float1} + {string.Join(", ", input.Float4)}";
+            }
+        }
+
+        [Fact]
+        public void TestCustomTransformer()
```

Comment:
You should also have a test that saves a model file with a CustomMappingTransformer, and then reads it back in, and executes it correctly.

Reply:
This is what `TestEstimatorCore` is doing underneath.

---
In reply to: [229415366](https://github.com/dotnet/machinelearning/pull/1406#discussion_r229415366) [](ancestors = 229415366)

Reply:
`TestEstimatorCore` doesn't execute the transform, from what I can tell.

Also, I think it should be read back in using a separate MLContext instance.

---
In reply to: [229416855](https://github.com/dotnet/machinelearning/pull/1406#discussion_r229416855) [](ancestors = 229416855,229415366)

Reply:
It surely executes the transformer, that's how I found bugs in it. See `CheckSame*` calls inside.

As for separate contexts, sure.

---
In reply to: [229419089](https://github.com/dotnet/machinelearning/pull/1406#discussion_r229419089) [](ancestors = 229419089,229416855,229415366)